### DISCUSSION TITLE
Global PWA for a NX-Workspace

### DISCUSSION THREAD

**Discussion by sanjeeb-KL** [nrwl/nx#13095] [General]
Hi Nx Team , I am using nx workspace for angular projects. In my nx workspace there are multiple projects I have added. When I am trying to install PWA for that It is installing to each project .But I want a global PWA for complete workspace. Whatever projects are there inside the Workspace it will come under one PWA , basically one Root level PWA a for all the projects not individual projects ?

**Comment by covik** [comment]
Are these 'projects' angular applications living independently of each other (ie. each one lives on its own (sub)domain)?

If that's the case you cannot have a root PWA since PWA manifests are scoped per (sub)domain. Multiple applications require multiple manfiests.

If by 'projects' you mean a single angular application backed by multiple libraries simply install PWA by telling the cli used to generate files or manually scaffold manifest and service worker in the application directory. Libraries should not care about PWA.

**Reply by sanjeeb-KL** [reply]
Hi @Covik ,Thanks for the reply .  In a single domain all those are deployed (i.e --- Let suppose domain is  **www.example.com** and projects are suppose A , B , C inside a nx workspace . To access Project A  we need hit  **www.example.com/A** similarly for B we need to hit **www.example.com/B**  same for C also.

So now I have added the service worker in A project , B project ...etc . But to run PWA for entire example.com it is not possible. So we have to install **example.com/A** , **example.com/B** separately . But I am searching for a solution to enable PWA for entire **example.com**  workspace not individual project.

**Reply by covik** [reply]
Your problem has nothing to do with NX and you might have a lot of work to do in order to support what you described here.

You will probably have more answers by explaining your problem on sites like stack overflow, reddit (r/javascript or r/webdev).

Maybe you can start [here](https://stackoverflow.com/questions/42474908/using-multiple-javascript-service-workers-at-the-same-domain-but-different-folde)?

**Comment by sanjeeb-KL** [comment]
Hii , Can u please suggest some more idea , how to achieve that global PWA  ?
